### DISCUSSION TITLE
otel.Tracer(...) concurrent usage

### DISCUSSION THREAD

**Discussion by schmurfy** [open-telemetry/opentelemetry-go#1780] [Q&A] [answered]
Sorry that is not an issue but I cannot find the answer and don't know where to ask: can the tracer object returned by `otel.Tracer("xxx")` be used in multiple goroutines or should I create it again when I need it ?

I did not found any clear documentation on this and I found conflicting examples where some declare a global variable with the tracer and reuse it and some call `otel.Tracer` each time they need to create a span.

**Comment by MrAlias** [comment, marked as answer]
[As specified](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#concurrency):

> Tracer - all methods are safe to be called concurrently.

Also, documented here in this project:

https://github.com/open-telemetry/opentelemetry-go/blob/23422c56df5578ae4f70f8bd5c0225dbad6be388/trace/trace.go#L671

The `otel.Tracer` function will return either an [internal implementation](https://github.com/open-telemetry/opentelemetry-go/blob/23422c56df5578ae4f70f8bd5c0225dbad6be388/internal/global/trace.go#L119), which is tested [here](https://github.com/open-telemetry/opentelemetry-go/blob/23422c56df5578ae4f70f8bd5c0225dbad6be388/internal/global/trace_test.go#L105) for concurrency safety, or the registered SDK version. If the default SDK, provided by this project, is used it as well is concurrent safe as tested [here](https://github.com/open-telemetry/opentelemetry-go/blob/23422c56df5578ae4f70f8bd5c0225dbad6be388/oteltest/harness.go#L221).

**Comment by craigpastro** [comment]
👋 A kind of continuation of this discussion I think: If we only plan to use one Tracer in a project, it is better, or perhaps more idiomatic say, to use `otel.Tracer(name)` where we need access to the tracer, or should the `tracer` be passed around?

I suppose I am more partial to just calling `otel.Tracer(name)` where needed as it simplifies my signatures.

Thanks very much!

**Reply by dmathieu** [reply]
Either works, though calling `otel.Tracer` will have a performance impact on heavily concurrent applications, as a mutex lock needs to be acquired.
See https://github.com/open-telemetry/opentelemetry-go/blob/23422c56df5578ae4f70f8bd5c0225dbad6be388/sdk/trace/provider.go#L110-L131

**Reply by craigpastro** [reply]
Hi @dmathieu ! Ah, that is very helpful to know. Thank you very much!
